price-estimator/src/error.rs

```diff
@@ -0,0 +1,14 @@
+//! Module implementing a custom warp rejection for internal service errors.
+
+use anyhow::Error;
+use warp::reject::{self, Reject, Rejection};
+
+#[derive(Debug)]
+pub struct InternalError(pub Error);
+
+impl Reject for InternalError {}
+
+/// Create a
+pub fn internal_server_rejection(err: Error) -> Rejection {
+    reject::custom(InternalError(err))
+}
```

Comment:
Might want to integrate this with the reject handler in `filter.rs`. I think what happens now is that the handler goes into the "unexpected internal error" branch. Or that is what you intend, just mentioning.

Reply:
#1141